Cookbook/Technical-Documents/AutomateAUIFlow.md

```diff
@@ -0,0 +1,259 @@
+
+We can also group common actions so long as they are within the scope of the screen and make logical sense, staying with the login screen we could have: 
+
+```swift 
+func enterCredentialsAndLogin(email: String)
+```
+
+#### Accessibility identifiers
+The historical method for handling Accessibility identifiers that we're going to be used in locating an **XCUIElement** was to define and stored within the screen object, normally in a *enum* for each element type. And then referenced from the functions using ```Cells.passwordField```. This ensured that identifiers are only stored once in test code and can be updated with a single change.
+
+This however is now evolved, with rather than having Accessibility identifiers define in production code and again in the Screen Object, we now access the identifier defined in a **AccessibilityContent** enum in the production code and reference from the automation code. ```AccessibilityContent.SignUp.termsAndConditionsCell``` this ensures identifiers are only defined and read from one location. Though is a work in progress.
+
+A detailed naming convention for accessibility identifiers can be found [here](https://github.com/Babylonpartners/ios-playbook/blob/master/Cookbook/Technical-Documents/AutomationIdentifiers.md).
+
+### API Calls
+When not using mocking, We use real API requests to control the state of the application. For example using the client API in the family appointment tests to add family member or the clinical API to prescribe medication. By completing these actions at the API level it removes the requirement to complete them through the UI, reducing execution time, removing duplication and facilitating tests that start in a controlled state. These network requests are normally handled by extending `APIInterface` with functional specific requests for example `APIInterface+Appointment`
+
+# Automate a flow
+We will now go through a set of truncated steps to automation a UI flow. Below is the example test scenario we intend to add.
+
+```swift
+    func test_login_as_uk_user() {
+        Given("I tap the login option in the Start up screen")
+        And("I enter credentials for a \"United Kingdom\" user")
+        And("the password field is secure")
+        When("I tap on the password visible icon")
+        Then("the password field is not secure")
+        When("I tap the login option in the Log in screen")
+        Then("the home screen is displayed")
+    }
+```
+## Feature
+We need to choose a *feature file* for the test scenario, for this example we will use `LoginFeature.swift` which already contains a number of sign in tests. Please **note** *that if you add a new Feature file, Xcode will automatically add the file to the default scheme causing it to be included in a Unit Test run, you will have to manually exclude it, or your PR will fail*. To begin with copy the test into the feature file ensuring you follow the `XCTest-Gherkin` formatting otherwise the code will fail to compile.
```

Comment:
Maybe this note should have its own exception, as it's very important information and will be a common question for developers that haven't wrote UI tests for our app yet.